source/Octopus.Tentacle/Versioning/AssemblyExtensions.cs

```diff
@@ -9,13 +9,13 @@ public static class AssemblyExtensions
         public static string GetFileVersion(this Assembly assembly)
         {
             var fileVersionInfo = FileVersionInfo.GetVersionInfo(assembly.FullLocalPath());
-            return fileVersionInfo.FileVersion;
+            return fileVersionInfo.FileVersion ?? "";
```

Comment:
Is "" a good backup value?

Reply:
Good question. We could return "0.0.0". It would be more friendly to the semantic version parser. But it would also hide the fact that there is no version. Thinking about the responsibility of this method, _should_ it know what a good default is? Maybe we should simply return a `string?` and let the consumer decide what they want to use as the default. For example

```csharp
// SemanticVersionInfo.cs
SemanticVersion = SemanticVersion.Parse(assembly.GetInformationalVersion() ?? "0.0.0");

// OctopusTentacle.cs
public static readonly string InformationalVersion = AssemblyExtensions.GetInformationalVersion(Assembly) ?? "Unknown";
```